### PR TITLE
Respect port configurations and support public/internal ports

### PR DESCRIPTION
**Pull request by dokipen**
Hello @ddbenson, @dnelson, @silverlyra, @micahnoland, 

Please review the following commits I made in branch 'dokipen/20190306180713-ports':

- **Respect port configurations and support public/internal ports** (8bcb14ad73871257fd7345a8f35a96dae78dc618)

- **Fixes resource tracking metadata** (4f306a1d5d564c4e398777b12ca5addfe174c781)

- **Update incarnation name** (50d696e89e05656300d0707c4b60c04f6713a5d0)

- **Use - namespaces** (2dfac42207c9a9466239a5f837d11e66e9437452)

- **Rename ReleaseManager** (ffbe8593c21410311cfd775d49469730e3149491)

- **Cleans code** (9fa6e2c2f76efdaeb9e79b6cc3e3246819b8e2a1)

R=@ddbenson
R=@dnelson
R=@silverlyra
R=@micahnoland

### CODE REVIEW

File: pkg/controller/releasemanager/releasemanager_controller.go [hunk 1]
```diff
@@ -147,21 +147,21 @@ func (r *ReconcileReleaseManager) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{Requeue: true}, nil
 }
 
-type ReleaseManager struct {
+type ResourceSyncer struct {
```
Comment (silverlyra) [inline pkg/controller/releasemanager/releasemanager_controller.go]: 👍 ty

File: pkg/controller/releasemanager/releasemanager_controller.go [hunk 2]
```diff
@@ -373,15 +378,36 @@ func (r *ReleaseManager) SyncVirtualService() error {
 		if current > 0 {
 			for _, port := range incarnation.Spec.Ports {
 				portNumber := uint32(port.Port)
-				releaseRoute, ok := releaseRoutes[portNumber]
+				filterHosts := port.Hosts
+				gateway := []string{"mesh"}
+				switch port.Mode {
+				case picchuv1alpha1.PortPublic:
+					if publicGateway == "" {
+						log.Info("Can't configure publicGateway, undefined in Cluster")
```
Comment (silverlyra) [inline pkg/controller/releasemanager/releasemanager_controller.go]:
```suggestion
						log.Warn("Can't configure publicGateway, undefined in Cluster")
```

at least

File: pkg/controller/releasemanager/releasemanager_controller.go [hunk 3]
```diff
@@ -373,15 +378,36 @@ func (r *ReleaseManager) SyncVirtualService() error {
...
+						continue
+					}
+					gateway = []string{publicGateway}
+					portNumber = uint32(port.IngressPort)
+				case picchuv1alpha1.PortPrivate:
+					if privateGateway == "" {
+						log.Info("Can't configure privateGateway, undefined in Cluster")
```
Comment (silverlyra) [inline pkg/controller/releasemanager/releasemanager_controller.go]:
```suggestion
						log.Warn("Can't configure privateGateway, undefined in Cluster")
```

File: pkg/controller/releasemanager/releasemanager_controller.go (hunk 2 above)
Comment (dokipen) [reply pkg/controller/releasemanager/releasemanager_controller.go]: https://godoc.org/github.com/go-logr/logr#InfoLogger

**Pull request merged**